src/index.js

```diff
@@ -4,11 +4,20 @@ import "./index.css";
 import App from "./App";
 import reportWebVitals from "./reportWebVitals";
 import { ModalProvider } from "./Context/ModalContext";
+import { PageProvider } from "./Context/PageContext";
+import { CartProvider } from "./Context/CartContext";
+import { WishlistProvider } from "./Context/WishlistContext";
 
 ReactDOM.render(
   <React.StrictMode>
     <ModalProvider>
-      <App />
+      <PageProvider>
```

Comment:
Its your choice, but you can create another component called maybe `PebbleEcomProviders` and add all these provider there and use this component in `index.js`. It will make it look cleaner.

Reply:
Creating a single Provider context would make that file bigger from a readability point of view, that's why I have created these many. In the future will surely look into creating a single provider.